Add `tk` commands to manage typeclasses

XVilka opened the pull request:
<!-- Filling this template is mandatory -->

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Expose typeclass information via the new `tk` commands.

**Test plan**

CI is green

**Closing issues**

Closes https://github.com/rizinorg/rizin/issues/3371

XVilka committed:
Add `tk` commands to manage typeclasses

ret2libc commented on librz/core/cmd_descs/cmd_type.yaml:
```diff
@@ -361,3 +361,29 @@ commands:
         cname: type_xrefs_list_all
         summary: List all types used by any function
         args: []
+  - name: tk
```
What are *typeclasses*? Please add a "Explanation" section in these commands so the user can get an idea.

ret2libc commented on librz/core/cmd/cmd_type.c:
```diff
@@ -798,3 +798,76 @@ RZ_IPI RzCmdStatus rz_type_xrefs_list_all_handler(RzCore *core, int argc, const
+		RZ_LOG_ERROR("\"%s\" type is not atomic\n", argv[1]);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	RzTypeTypeclass tclass = rz_type_typeclass_from_string(argv[2]);
+	rz_cons_println(rz_type_typeclass_as_string(tclass));
+	return RZ_CMD_STATUS_OK;
```
Where is the typeclass *set* here?

XVilka replied:
It's a draft pull request, no need to review just yet. I am working on "set" since it can be complicated because it need to propagate all over the chain of typedefs, etc

XVilka committed:
type: add `tk` commands to list and manage typeclasses

Expose the typeclasses of types through the new `tk` commands:

- `tk <type>` shows the typeclass of a type;
- `tkl` lists the available typeclasses, while `tkll` (verbose), `tklt`
  (table) and `tklj` (JSON) additionally list the types belonging to each
  typeclass;
- `tks <type> <typeclass>` sets the typeclass of a type.

A new rz_base_type_set_typeclass() API backs the `tks` command. Since a
typedef without its own typeclass inherits the one of the type it points
to, setting the typeclass of an atomic type is automatically reflected on
the typedefs resolving to it.

The `tk` help also explains what typeclasses are and lists the available
ones.

Closes #3371
